rust-runtime/aws-smithy-runtime/tests/stalled_stream_upload.rs

```diff
@@ -107,27 +107,85 @@ async fn upload_stalls() {
     expect_timeout(result.await.expect("no panics"));
 }
 
+/// Scenario: Request does not have a body. Server response doesn't start coming through
+///           until after the grace period.
+/// Expected: MUST NOT timeout.
+#[tokio::test]
+async fn empty_request_body_delayed_response() {
+    let _logs = show_test_logs();
+
+    let (server, time, sleep) = stalling_server(Some(Duration::from_secs(6)));
+    let op = operation(server, time.clone(), sleep);
+
+    let result = tokio::spawn(async move { op.invoke(SdkBody::empty()).await });
+
+    let _advance = tokio::spawn(async move {
+        for _ in 0..6 {
+            tick!(time, Duration::from_secs(1));
+        }
+    });
+
+    assert_eq!(200, result.await.unwrap().expect("success").as_u16());
+}
+
 /// Scenario: All the request data is either uploaded to the server or buffered in the
 ///           HTTP client, but the response doesn't start coming through within the grace period.
-/// Expected: MUST timeout after the grace period completes.
+/// Expected: MUST NOT timeout, upload throughput should only apply up until the request body has
+/// been read completely and handed off to the HTTP client.
 #[tokio::test]
-async fn complete_upload_no_response() {
+async fn complete_upload_delayed_response() {
     let _logs = show_test_logs();
 
-    let (server, time, sleep) = stalling_server();
+    let (server, time, sleep) = stalling_server(Some(Duration::from_secs(6)));
     let op = operation(server, time.clone(), sleep);
 
     let (body, body_sender) = channel_body();
     let result = tokio::spawn(async move { op.invoke(body).await });
 
     let _streamer = tokio::spawn(async move {
+        info!("send data");
         body_sender.send(NEAT_DATA).await.unwrap();
         tick!(time, Duration::from_secs(1));
+        info!("body send complete; dropping");
         drop(body_sender);
-        time.tick(Duration::from_secs(6)).await;
+        // advance by the grace period
+        tick!(time, Duration::from_secs(5));
```

Comment:
```suggestion
        tick!(time, DEFAULT_GRACE_PERIOD);
```
after making `DEFAULT_GRACE_PERIOD` available in the test

Reply:
ty Yuki, good catch